src/main/core/manager.c

```diff
@@ -409,9 +442,40 @@ static gchar** _manager_generateEnvv(Manager* manager, InterposeMethod interpose
             if (key != NULL && value != NULL) {
                 /* check if the key is LD_PRELOAD */
                 if (!g_ascii_strncasecmp(key, "LD_PRELOAD", 10)) {
-                    /* append all LD_PRELOAD entries */
-                    debug("adding key path %s", value);
-                    g_ptr_array_add(ldPreloadArray, g_strdup(value));
+                    // Handle the list of entries, which could be separated by ' ' or ':'.
+                    GString* preloadBuf = g_string_new(value);
+
+                    // First replace all occurences of ' ' by ':'.
+                    // g_string_replace does what we want, but isn't available until glib 2.68.
+                    for (int i = 0; i < preloadBuf->len; i++) {
+                        if (preloadBuf->str[i] == ' ') {
+                            preloadBuf->str[i] = ':';
+                        }
+                    }
+
+                    // Now split by ':' to handle each preload path.
+                    gchar** paths = g_strsplit(preloadBuf->str, ":", 0);
+
+                    // Expand any paths that begin with '~'.
+                    for (int i = 0; paths != NULL && paths[i] != NULL; i++) {
+                        GString* pathBuf = g_string_new(NULL);
+
+                        if (!g_ascii_strncasecmp(paths[i], "~/", 2)) {
```

Comment:
Nit - I don't know of a benefit to hard-coding the string length when one of the strings being compared is hard-coded; the comparison will stop when the end of one of the strings is reached. i.e. I think this can be `if (!g_ascii_strcasecmp(paths[i], "~/")) {`

Reply:
Done in ff4d95aedfe270fd40c18317504f919c7216d70a

Reply:
I think the number is needed so that it only compares the first characters. For example I think `g_ascii_strncasecmp("~/test", "~/", 2) == 0`, but `g_ascii_strcasecmp("~/test", "~/") > 0`.

Reply:
Oh, of course you're right. Sorry, my mistake!

I think g_str_has_prefix is the nicer thing to use here? https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-str-has-prefix

Reply:
(I unresolved this since I think we want to change it back before merging.)

Reply:
Good catch!

I tried again in 2e8d80a447c638cf38008a50371c14b489e8668a

Reply:
Looks good to me.

Reply:
Hmm, it looks like unresolving did not stop the auto-merge.